TACTSharp/EncodingInstance.cs

```diff
@@ -86,158 +83,118 @@ public bool TryGetEKeys(Span<byte> cKeyTarget, out EncodingResult? result)
-            var offs = 0;
-            while (true)
+            ReadOnlySpan<byte> fileData = new(pageData, (int) new FileInfo(_filePath).Length);
+            var targetPage = _header.CEKey.ResolvePage(fileData, cKeyTarget);
+            while (targetPage.Length != 0)
             {
-                if (offs >= eKeyPageSize)
-                    break;
+                var keyCount = targetPage[0];
+                var recordData = targetPage.Slice(1, 5 + _header.CKeySize + _header.EKeySize * keyCount);
 
-                var eKeyCount = targetPage[offs];
-                offs++;
+                // Advance iteration
+                targetPage = targetPage[(recordData.Length + 1) ..];
 
-                if (eKeyCount == 0)
+                if (keyCount == 0)
                     continue;
 
-                if (targetPage.Slice(offs + 5, header.hashSizeCKey).SequenceEqual(cKeyTarget))
-                {
-                    var decodedFileSize = (ulong)targetPage.Slice(offs, 5).ReadInt40BE();
-                    offs += 5;
-
-                    offs += header.hashSizeCKey; // +ckey
+                var recordContentKey = recordData.Slice(5, _header.CKeySize);
+                var recordEncodingKeys = recordData.Slice(5 + _header.CKeySize, _header.EKeySize * keyCount);
 
-                    var eKeys = new List<byte[]>(eKeyCount);
-                    for (var i = 0; i < eKeyCount; i++)
-                    {
-                        var eKey = new byte[header.hashSizeEKey];
-                        targetPage.Slice(offs, header.hashSizeEKey).CopyTo(eKey);
-                        offs += header.hashSizeEKey;
-                        eKeys.Add(eKey);
-                    }
-
-                    return new EncodingResult()
-                    {
-                        eKeyCount = eKeyCount,
-                        decodedFileSize = decodedFileSize,
-                        eKeys = eKeys
-                    };
-                }
-                else
+                if (recordContentKey.SequenceEqual(cKeyTarget))
                 {
-                    offs += 5; //+size
-                    offs += header.hashSizeCKey; // +ckey
-                    offs += header.hashSizeEKey * eKeyCount; // +ekeys
+                    var decodedFileSize = (ulong)recordData.ReadInt40BE();
+                    return new EncodingResult(keyCount, recordEncodingKeys, decodedFileSize);
                 }
             }
 
-            return null;
+            return EncodingResult.Empty;
         }
 
-        public unsafe (string eSpec, ulong encodedFileSize)? GetESpec(Span<byte> eKeyTarget)
+        public unsafe (string eSpec, ulong encodedFileSize) GetESpec(Span<byte> eKeyTarget)
         {
             byte* pageData = null;
             mmapViewHandle.AcquirePointer(ref pageData);
-            lock (ESpecLock)
+
+            ReadOnlySpan<byte> fileData = new(pageData, (int) new FileInfo(_filePath).Length);
```

Comment:
`stat()` again.

Reply:
There's a couple ways to go about this.

* Store the pointer + file size in `EncodingInstance`, which means this now needs to implement `IDisposable`. 
* Just store size and keep asking for a pointer whenever needed, assuming the file never changes for the lifetime of the program.

Technically the second option also needs `IDisposable` because memory mapped files implement `IDisposable`.

Reply:
Nevermind that, blindly trust decompressed size from config.